lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -676,36 +676,158 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
       builder.create<ttkernel::BinaryOpInitCommonOp>(
           op.getLoc(), cbOperands[operandIndices[0]],
           cbOperands[operandIndices[1]], cbOperands[numDpsInputs]);
-      builder.create<ttkernel::MulTilesInitOp>(op.getLoc(),
-                                               cbOperands[operandIndices[0]],
-                                               cbOperands[operandIndices[1]]);
+      builder.create<ttkernel::MulTilesInitFOp>(op.getLoc());
+    } else if(mlir::isa<arith::DivFOp>(op)) {
+      builder.create<ttkernel::BinaryOpInitCommonOp>(
+          op.getLoc(), cbOperands[operandIndices[0]],
+          cbOperands[1], cbOperands[numDpsInputs]);
+      builder.create<ttkernel::MulTilesInitFOp>(op.getLoc());
     } else {
       llvm_unreachable("Unhandled conversion");
     }
   }
 
+  void commonComputeMulOp(Operation &op, ArrayRef<BlockArgument> cbOperands,
+                          ArrayRef<BlockArgument> iterators,
+                          SmallVector<unsigned> blockArgIteratorMapping,
+                          OpBuilder &builder, Block *computeBlock,
+                          std::int64_t numDpsInputs,
+                          SmallVector<std::int64_t>& operandIndices) const {
+    Value dstIndex = i32(0, builder);
+    Value output = cbOperands[numDpsInputs];
+    builder.create<ttkernel::TileRegsAcquireOp>(
+      op.getLoc());
+    if (mlir::isa<arith::MulFOp>(op)) {
+      builder.create<ttkernel::MulTilesOp>(
+        op.getLoc(), cbOperands[operandIndices[0]],
+        cbOperands[operandIndices[1]], iterators[blockArgIteratorMapping[0]],
+        iterators[blockArgIteratorMapping[0]], dstIndex);
+    } else if (mlir::isa<arith::DivFOp>(op)) {
+      builder.create<ttkernel::MulTilesOp>(
+        op.getLoc(), cbOperands[operandIndices[0]],
+        cbOperands[operandIndices[1]], iterators[blockArgIteratorMapping[0]],
+        dstIndex, dstIndex);
+    } else {
+      llvm_unreachable("Common compute for multiplying tiles should be called only on MulFOp and DivFOp");
+    }
+    
+    builder.create<ttkernel::TileRegsCommitOp>(
+      op.getLoc());
+    builder.create<ttkernel::TileRegsWaitOp>(
+      op.getLoc());
+    builder.create<ttkernel::PackTileOp>(
+      op.getLoc(), dstIndex, output,
+      iterators[blockArgIteratorMapping[numDpsInputs]]);
+    builder.create<ttkernel::TileRegsReleaseOp>(
+      op.getLoc());
+  }
+
+  void commonComputeRecipOp(Operation &op, ArrayRef<BlockArgument> cbOperands,
+                            ArrayRef<BlockArgument> iterators,
+                            SmallVector<unsigned> blockArgIteratorMapping,
+                            OpBuilder &builder, Block *computeBlock,
+                            std::int64_t numDpsInputs,
+                            SmallVector<std::int64_t>& operandIndices) const {
+    Value dstIndex = i32(0, builder);
+    Value one = i32(1, builder);
+    
+    builder.create<ttkernel::CopyTileToDstInitShort>(
```

Comment:
I think it is similar, since it initializes packer in some way, we need to call it before every `copy_tile`. Workload hangs when init is before for loop